### DISCUSSION TITLE
Get id's of all selected items

### DISCUSSION THREAD

**Discussion by LickABrick** [bryanmylee/svelte-headless-table#188] [Q&A]
Hi,

I have followed the steps described here to create a data-table: https://www.shadcn-svelte.com/docs/components/data-table#row-selection

In my data-table the 
A question I have is how do I get the id's from all selected items in the table? With id I mean the values from my 'DB'.
In the example this is the Payments array: https://www.shadcn-svelte.com/docs/components/data-table#initialize-table

**Comment by bryanmylee** [comment]
Seems like that tutorial uses the [`addSelectedRows`](https://svelte-headless-table.bryanmylee.com/docs/plugins/add-selected-rows) plugin, which exposes its plugin state like so:

```svelte
<script>
  const table = createTable(data, {
    select: addSelectedRows({...}),
  });
  const columns = table.createColumns(...);
  const {headerRows, rows, pluginStates} = table.createViewModel(columns);
  const {selectedDataIds} = pluginStates.select;
  $: console.log($selectedDataIds);
</script>
```

**Reply by bwprado** [reply]
@bryanmylee, do you know how would we be able to get the data from the selected rows? Could not find an example of that.

**Comment by LickABrick** [comment]
Sorry, I actually searched issues instead of discussions before posting this question. I fixed it using the answer in: #110

**Reply by patra0o** [reply]
Hi mate, would you mind sharing a snippet?
